rados/snapshot.go

```diff
@@ -0,0 +1,72 @@
+package rados
+
+// #cgo LDFLAGS: -lrados
+// #include <stdlib.h>
+// #include <rados/librados.h>
+import "C"
+
+import "unsafe"
+
+// Snapshot represents a snapshot on the particular ioctx.
+type Snapshot struct {
+	ioctx *IOContext
+	name  string
+}
+
+// CreateSnapshot creates a pool-wide snapshot.
+//
+// Implements:
+//  int rados_ioctx_snap_create(rados_ioctx_t io, const char *snapname)
+func (ioctx *IOContext) CreateSnapshot(snapName string) (*Snapshot, error) {
+	if ioctx.ioctx == nil {
+		return nil, ErrNoIOContext
+	}
+
+	cSnapName := C.CString(snapName)
```

Comment:
you should probably check if `snapName != ""`

Reply:
Actually, ceph doesn't have a problem if snap name is " " 
Same is the case with rbd snapshot also, let me know if we should ban that from go-ceph side.

Reply:
oh, if Ceph is okay with it, we should not block it either